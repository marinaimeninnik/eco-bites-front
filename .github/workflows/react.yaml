name: Deploy to S3 and Invalidate CloudFront

on:
  push:
    branches:
      - main
      - CI-CD_pipeline
  workflow_dispatch:
  workflow_call:

jobs:
  deploy:
    permissions:
        id-token: write
        contents: read
    runs-on: ubuntu-latest

    steps:
    - name: Checkout
      uses: actions/checkout@v2

    - name: Configure AWS credentials
      uses: aws-actions/configure-aws-credentials@v4
      with:
        role-to-assume: ${{ vars.ROLE_ARN }}
        role-session-name: github-actions-labsadm
        aws-region: ${{ vars.AWS_REGION }}

    - name: Download artifact
      uses: actions/download-artifact@v4
      with:
        name: tf-outputs
        repository: marinaimeninnik/eco-bites-back
       
    - name: Load instance ID from file
      run: cat tf-outputs.env >> $GITHUB_ENV

    - name: Check variables
      run: echo 'bucket_name is ${{ env.FRONTEND_S3_NAME }}'

    - name: Build frontend
      run: npm run build

    # - name: Deploy to S3
    #   run: aws s3 sync ./build s3://${{ env.FRONTEND_S3_NAME }}
    # # env:
    # #     AWS_S3_BUCKET: ${{ secrets.AWS_S3_BUCKET }}

    # - name: Invalidate CloudFront Distribution
    #   uses: chetan/invalidate-cloudfront-action@v1
    #   env:
    #     DISTRIBUTION: ${{ env.CF_ID }}
    #     PATHS: '/*'
    #     #   AWS_REGION: 'us-west-2'
    #     #     AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
    #     #     AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}