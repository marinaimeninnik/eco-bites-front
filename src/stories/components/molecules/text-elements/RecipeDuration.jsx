import PropTypes from 'prop-types';
import Time from '../../atomic/text-elements/Time';
import ClockIcon from '../../atomic/icons/ClockIcon';
import { Stack } from '@mui/system';

const RecipeDuration = ({ typographyVariant, minutes, showClock }) => (
    <Stack
        sx={{ mb: 1 }}
        direction="row"
        display={'flex'}
        alignItems={'center'}
        gap={showClock ? 0.5 : 0}
    >
        {showClock ? <ClockIcon size="medium" /> : null}
        <Time minutes={minutes} variant={typographyVariant} />
    </Stack>
);

RecipeDuration.defaultProps = {
    typographyVariant: 'Emphasis/E12',
    showClock: true,
};

RecipeDuration.propTypes = {
    typographyVariant: PropTypes.oneOf(['Paragraph/P12', 'Emphasis/E12']),
    minutes: PropTypes.number.isRequired,
    showClock: PropTypes.bool,
};

export default RecipeDuration;
