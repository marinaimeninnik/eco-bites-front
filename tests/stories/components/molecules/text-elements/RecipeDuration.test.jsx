import { render } from '@testing-library/react';
import RecipeDuration from 'src/stories/components/molecules/text-elements/RecipeDuration';

describe('RecipeDuration Component', () => {
    it('renders correctly', () => {
        const { asFragment } = render(<RecipeDuration minutes={60} />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('renders without crashing', () => {
        const { getByText } = render(<RecipeDuration minutes={60} />);
        expect(getByText('1h')).toBeInTheDocument();
    });

    it('shows ClockIcon when showClock is true', () => {
        const { getByTestId } = render(
            <RecipeDuration showClock={true} minutes={60} />
        );
        expect(getByTestId('clock-icon')).toBeInTheDocument();
    });

    it('does not show ClockIcon when showClock is false', () => {
        const { queryByTestId } = render(
            <RecipeDuration showClock={false} minutes={60} />
        );
        expect(queryByTestId('clock-icon')).toBeNull();
    });

    it('renders Time component with correct props', () => {
        const { container } = render(<RecipeDuration minutes={60} />);
        expect(container.querySelector('time').getAttribute('datetime')).toBe(
            'PT60M'
        );
    });
});
