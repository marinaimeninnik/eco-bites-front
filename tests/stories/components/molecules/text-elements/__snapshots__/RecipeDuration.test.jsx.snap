// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeDuration Component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiStack-root css-1apgyqu-MuiStack-root"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="clock-icon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2m3.3 14.71L11 12.41V7h2v4.59l3.71 3.71z"
      />
    </svg>
    <time
      class="MuiTypography-root MuiTypography-Emphasis/E12 css-37jdci-MuiTypography-root"
      datetime="PT60M"
    >
      1h
    </time>
  </div>
</DocumentFragment>
`;
